### ISSUE TITLE
Combobox `inline` mode has unexpected dropdown keyboard navigation

### ISSUE THREAD

**Issue by atmgrifter00** [ni/nimble#1985]
# 🐛 Bug Report

When navigating the dropdown of a `Combobox` that has its `autocomplete` set to `inline`, then if the `Combobox` has its value set to something other than '', it will only navigate between items in the dropdown that match from the beginning of the current value. For example, if the value is "Sue", and there are "Sue", "Mary", and "Sue Ann" in the dropdown, a user can only navigate between "Sue" and "Sue Ann" in the dropdown, despite "Mary" being visible.

## 💻 Repro or Code Sample

This is an issue in the [FAST `Combobox` implementation](https://explore.fast.design/components/fast-combobox).

## 🤔 Expected Behavior

All dropdown options should be able to be selected via keyboard navigation.

## 😯 Current Behavior

As described, if a value has been set, keyboard navigation will skip over all options that don't match the current value text from the beginning.

## 💁 Possible Solution

The problem almost certainly  has to do with the `Combobox` relying on its `filteredOptions` which are not being reset to the entire available options, when the `autocomplete` isn't set to a filter mode (either `list` or `both`).

## 🌍 Your Environment

* OS & Device: [e.g. MacOS, iOS, Windows, Linux] on [iPhone 7, PC]
* Browser [e.g. Microsoft Edge, Google Chrome, Apple Safari, Mozilla FireFox]
* Version [e.g. 1.8.0]
